### DISCUSSION TITLE
Support hass agent media

### DISCUSSION THREAD

**Discussion by MeowRay** [music-assistant/support#976] [Feature requests and ideas]
https://github.com/LAB02-Research/HASS.Agent-Integration

**Comment by LAB02-Admin** [comment]
Was actually just posting a ticket for this when I saw your request, thanks @MeowRay :) 

Some background:

I'm the developer of HASS.Agent, a companion app for Windows. It has mediaplayer functionality, and uses an integration on HA's end which creates a `media_player` entity.

Unfortunately, MA doesn't recognize it and I'm not sure why. Could you perhaps take a look at the mediaplayer code (it's nothing fancy) to see what I've missed?

https://github.com/LAB02-Research/HASS.Agent-Integration/blob/main/custom_components/hass_agent/media_player.py

I followed this:

https://github.com/music-assistant/hass-music-assistant/discussions/622

Sending mp3 files via the standard media browser works just fine.

Note: also tried with the MA's latest v2022.11.0 beta.

**Reply by marcelveldt** [reply]
The media player needs to support volume_set feature.
That will be changed in the next (December) release where I've rewritten it all to be a bit for flexible.
For now workaround it by setting the volume_set feature

**Reply by LAB02-Admin** [reply]
Yep that does it, great. I'll implement volume control, good feature to have anyway.

Thanks for the quick reply.

**Comment by LAB02-Admin** [comment]
This is completed, requires version 2022.11.9 of the integration and [2022.14.0-beta1](https://github.com/LAB02-Research/HASS.Agent/releases/tag/2022.14.0-beta1) of HASS.Agent.
